gensim/models/keyedvectors.py

```diff
@@ -185,95 +183,97 @@
-        Possibly have `positive` and `negative` list of entities in `**kwargs`.
+    def _load_specials(self, *args, **kwargs):
+        super(KeyedVectors, self)._load_specials(*args, **kwargs)
+        # fixup rename/consolidation into index2key of older index2word, index2entity
+        if not hasattr(self, 'index2key'):
+            self.index2key = self.__dict__.pop('index2word', self.__dict__.pop('index2word', None))
 
-        """
-        return NotImplementedError()
+    def __getitem__(self, entities):
+        """Get vector representation of `entities`.
 
-    def distance(self, entity1, entity2):
-        """Compute distance between vectors of two input entities, specified by their string id."""
-        raise NotImplementedError()
+        Parameters
+        ----------
+        entities : {str, list of str}
+            Input entity/entities.
 
-    def distances(self, entity1, other_entities=()):
-        """Compute distances from a given entity (its string id) to all entities in `other_entity`.
-        If `other_entities` is empty, return the distance between `entity1` and all entities in vocab.
+        Returns
+        -------
+        numpy.ndarray
+            Vector representation for `entities` (1D if `entities` is string, otherwise - 2D).
 
         """
-        raise NotImplementedError()
+        if isinstance(entities, string_types):
+            # allow calls like trained_model['office'], as a shorthand for trained_model[['office']]
+            return self.get_vector(entities)
 
-    def get_vector(self, entity):
+        return vstack([self.get_vector(entity) for entity in entities])
+
+    def get_vector(self, key, use_norm=False):
         """Get the entity's representations in vector space, as a 1D numpy array.
 
         Parameters
         ----------
-        entity : str
-            Identifier of the entity to return the vector for.
+        key : str
+            Identifier of the vector to return
+        use_norm : bool, optional
+            If True - resulting vector will be L2-normalized (unit euclidean length).
 
         Returns
         -------
         numpy.ndarray
-            Vector for the specified entity.
+            Vector for the specified key.
 
         Raises
         ------
         KeyError
-            If the given entity identifier doesn't exist.
+            If the given key doesn't exist.
 
         """
-        if entity in self.vocab:
-            result = self.vectors[self.vocab[entity].index]
+        if key in self.vocab:
+            if use_norm:
+                result = self.vectors_norm[self.vocab[key].index]
+            else:
+                result = self.vectors[self.vocab[key].index]
             result.setflags(write=False)
             return result
         else:
-            raise KeyError("'%s' not in vocabulary" % entity)
+            raise KeyError("Key '%s' not in vocabulary" % key)
+
+    def word_vec(self, *args, **kwargs):
```

Comment:
```suggestion
    word_vec = get_vector  # Compatibility alias for get_vector()
```

What's the advantage of keeping the alias function standalone?

Reply:
Hmm, probably nothing - just how I got there incrementally. (Perhaps: to give it a doc-comment declaring it as a shim? Unsure how this assigment might affect doc-generation. Would `word_vec()` show an exact copy of `get_vector()` doc-comment?)

Reply:
Yes, it would. It's just an object alias.